src/plugins/hetero/executable_network.cpp

```diff
@@ -68,6 +70,11 @@ HeteroExecutableNetwork::HeteroExecutableNetwork(const InferenceEngine::CNNNetwo
       _config{config} {
     auto function = network.getFunction();
     IE_ASSERT(function != nullptr);
+
+    ngraph::pass::Manager manager;
+    manager.register_pass<ngraph::pass::ConstantFolding>();
+    manager.run_passes(std::const_pointer_cast<ov::Model>(function));
```

Comment:
Is it really safe to 'const_cast' here? Per my understanding, plugins can modify their 'executable network', but it is not expected from plugins to modify original `CNNNetwork`. Also 'multi' plugin can use multiple threads to load same `CNNNetwork` to different devices. If one of them modifies model, race condition can occur.
Can we probably apply this transformation on `clonedFunction`?

Reply:
Agree with @nosovmik Please clone model, we shouldn't change the original model